staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -136,7 +139,7 @@ func (pq waitForPriorityQueue) Peek() interface{} {
 // ShutDown gives a way to shut off this queue
 func (q *delayingType) ShutDown() {
 	q.Interface.ShutDown()
-	close(q.stopCh)
+	q.stopOnce.Do(func() { close(q.stopCh) })
```

Comment:
can we put all three calls inside stopOnce.Do? that makes it more obviously clear that this is safe to invoke multiple times. We should also document that all calls after the first one do nothing (similar to https://golang.org/pkg/context/#CancelFunc)

Reply:
sure

Reply:
I wish we'd called this method Close